### DISCUSSION TITLE
How to add friction between objects

### DISCUSSION THREAD

**Discussion by datou1209** [sofa-framework/sofa#4174] [Write / Run a simulation] [answered]
hello everyone，
I am using Sofa to create a simulation environment for my soft robot. I need to place it on the floor to achieve friction with the floor. However, after I added the friction, some problems occurred. When the inner cavity of the robot increased When under pressure, the robot's motion is inconsistent with common sense. Sometimes it will slide or stand up involuntarily. This is the first time I simulated it through Sofa. I don't know whether this is normal or abnormal.

Also, when I increase the distance between the floor and the robot, the robot goes through the floor and falls off the floor.

My sofa version is 23.06, which is compiled by myself using binary files. And my project address is [sofa_soft_robot](https://github.com/datou1209/sofa_soft_robot)

![image](https://github.com/sofa-framework/sofa/assets/127370625/a2b17b0c-0ddb-417f-98d4-9d187c6da1f2)

If i change the distance between the floor and robot，

![image](https://github.com/sofa-framework/sofa/assets/127370625/08606a5a-aa12-4046-843c-e179d1adfa30)

**Comment by bakpaul** [comment, marked as answer]
Hi datou ! 

I would like to help you but I will need to have a look at your scene file. I saw your project and there are many of them. 

For the friction itself you would need to use a collision pipeline. For a good example you can look at [fallingBeamLagrangianCollision.scn](https://github.com/sofa-framework/sofa/blob/master/examples/Demos/fallingBeamLagrangianCollision.scn)
